src/Core/Migration/V6_7/Migration1728040169AddPrimaryOrderDelivery.php

```diff
@@ -0,0 +1,70 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Migration\V6_7;
+
+use Doctrine\DBAL\ArrayParameterType;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\ParameterType;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Migration\MigrationStep;
+
+/**
+ * @internal
+ */
+#[Package('checkout')]
+class Migration1728040169AddPrimaryOrderDelivery extends MigrationStep
+{
+    public function getCreationTimestamp(): int
+    {
+        return 1728040169;
+    }
+
+    public function update(Connection $connection): void
+    {
+        // No foreign key set both from order -> (primary) order delivery on
+        // purpose so the DAL can handle the circular reference. We have a similar situation with the order and order
+        // address.
+        if (!$this->columnExists($connection, 'order', 'primary_order_delivery_id')) {
```

Comment:
The entire process is encapsulated in the 'columnExists' function. As this can be a long-running migration, wouldn't it make more sense to only alter the column conditionally and always loop?